app/scripts/controllers/metametrics.js

```diff
@@ -94,6 +104,28 @@ export default class MetaMetricsController {
       this.network = getNetworkIdentifier();
     });
     this.segment = segment;
+
+    // Track abandoned fragments that weren't properly cleaned up.
+    // Abandoned fragments are those that were stored in persistent memory
+    // and are available at controller instance creation, but do not have the
+    // 'persist' flag set. This means anytime the extension is unlocked, any
+    // fragments that are not marked as persistent will be purged and the
+    // failure event will be emitted.
+    Object.values(abandonedFragments).forEach((fragment) => {
+      this.finalizeEventFragment(fragment.id, { abandoned: true });
+    });
+
+    // Close out event fragments that were created but not progressed
+    setInterval(() => {
+      this.store.getState().fragments.forEach((fragment) => {
+        if (
+          fragment.timeout &&
+          Date.now() - fragment.lastUpdated / 1000 > fragment.timeout
+        ) {
+          this.finalizeEventFragment(fragment.id, { abandoned: true });
+        }
+      });
+    }, SECOND * 30);
```

Comment:
Not sure why above code requires `setInterval` ?

Reply:
Oops. sorry forgot to mention i added more docs.